### PR TITLE
Git hooks with pre-commit

### LINKED ISSUE #2592: Check commited files for lfs!
lfs'ing files after there were merged into master is always a pain...

Implement a check

### PR DESCRIPTION
**Pull request by bilke**
Adds git hooks via [pre-commit](https://pre-commit.com/) hooks manager.

Has to be manually enabled:

```
pip install pre-commit
cd ogs
pre-commit install
```

Currently checks for:

- Trailing whitespace (and auto-corrects them)
- Git LFS files to be correctly commited
- XML syntax check
- Yaml syntax check
- Large (non-lfs files), rejects files larger than 1 MB
- All file extensions have to be lower-case

Excluded directories: `ThirdParty`, `Tests/Data`

You can run it also manually on the currently staged files:

```
pre-commit run
```

Or on all files in the repo (this is also done in Jenkins):

```
$ pre-commit run --all-files
Trim Trailing Whitespace.................................................Passed
Check for added large files..............................................Passed
Check for merge conflicts................................................Passed
Check Xml................................................................Passed
Check Yaml...............................................................Passed
Run git diff --check.....................................................Passed
Check file extensions....................................................Passed
Check git lfs files......................................................Passed
```

Fixes #2592.

## TODO:

1. [x] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.2.1)
2. [x] Any new feature or behavior change was documented?
3. [x] Add hook to check for lower-case file extensions
3. [x] Does this work Windows?

### CODE REVIEW

File: scripts/docker/Dockerfile.gcc.full [hunk 1]
```diff
@@ -32,15 +32,15 @@ RUN apt-get update -y && \
         python3-setuptools \
         python3-wheel && \
     rm -rf /var/lib/apt/lists/*
-RUN pip3 install virtualenv
-# CMake version 3.13.4
+RUN pip3 install virtualenv pre-commit
+# CMake version 3.15.1
```
Comment (endJunction) [inline scripts/docker/Dockerfile.gcc.full]: Is 3.15.1 the new minimum for cmake or only in docker?

Comment (bilke) [reply scripts/docker/Dockerfile.gcc.full]: no, my fault. I now set this to 3.12.4.

**Comment by bilke** [comment]
I have not added the linked clang-format hook as it specifially works on the acutal git diff. The pre-commit tool works a bit differently: it gives the hooks scripts the filenames of the changed files or all files in the repo (when running with `--all-files`). Also `clang-format` does not return a value if it actually formatted the file. For the moment I am not sure how to properly integrate both together...

**Comment by endJunction** [comment]
:fast_forward:

**Pull request merged**